vault/identity_store_oidc.go

```diff
@@ -0,0 +1,743 @@
+				logical.UpdateOperation: i.handleOIDCCreateKey(),
+				logical.ReadOperation:   i.handleOIDCReadKey(),
+				logical.DeleteOperation: i.handleOIDCDeleteKey(),
+			},
+			HelpSynopsis:    "oidc/key/:key help synopsis here",
+			HelpDescription: "oidc/key/:key help description here",
+		},
+		{
+			Pattern: "oidc/key/" + framework.GenericNameRegex("name") + "/rotate/?$",
+			Fields: map[string]*framework.FieldSchema{
+				"name": &framework.FieldSchema{
+					Type:        framework.TypeString,
+					Description: "Name of the key",
+				},
+				"verification_ttl": &framework.FieldSchema{
+					Type:        framework.TypeString,
+					Description: "Controls how long the public portion of a key will be available for verification after being rotated. Setting verification_ttl here will override the verification_ttl set on the key.",
+				},
+			},
+			Callbacks: map[logical.Operation]framework.OperationFunc{
+				logical.UpdateOperation: i.handleOIDCRotateKey(),
+			},
+			HelpSynopsis:    "oidc/key/:key/rotate help synopsis here",
+			HelpDescription: "oidc/key/:key/rotate help description here",
+		},
+		{
+			Pattern: "oidc/key/?$",
+			Callbacks: map[logical.Operation]framework.OperationFunc{
+				logical.ListOperation: i.handleOIDCListKeys(),
+			},
+			HelpSynopsis:    "list oidc/key/ help synopsis here",
+			HelpDescription: "list oidc/key/ help description here",
+		},
+		{
+			Pattern: "oidc/well-known/certs/?$",
+			Callbacks: map[logical.Operation]framework.OperationFunc{
+				logical.ReadOperation: i.handleOIDCReadCerts(),
+			},
+			HelpSynopsis:    "read oidc/well-known/certs/ help synopsis here",
+			HelpDescription: "read oidc/well-known/certs/ help description here",
+		},
+	}
+}
+
+// handleOIDCCreateKey is used to create a new key
+func (i *IdentityStore) handleOIDCCreateKey() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+
+		var namedKey *NamedKey
+		var update bool = false
+		var algorithm signingAlgorithm
+
+		// parse parameters
+		name := d.Get("name").(string)
+		rotationPeriod := d.Get("rotation_period").(string)
+		verificationttl := d.Get("verification_ttl").(string)
+		algorithmInput := d.Get("algorithm").(string)
+
+		// determine if we are creating a key or updating an existing key
+		entry, err := req.Storage.Get(ctx, "oidc-config/named-key/"+name)
+		if err != nil {
+			return nil, err
+		}
+		if entry != nil {
+			if err := entry.DecodeJSON(&namedKey); err != nil {
+				return nil, err
+			}
+			update = true
+		}
+
+		// set defaults if creating a new key
+		if !update {
+			if rotationPeriod == "" {
+				rotationPeriod = "6h"
+			}
+			if verificationttl == "" {
+				verificationttl = rotationPeriod
+			}
+			if algorithmInput == "" {
+				algorithmInput = "RS256"
+			}
+		}
+
+		// verify inputs and set values if this is an update
+		if rotationPeriod != "" {
+			_, err = parseutil.ParseDurationSecond(rotationPeriod)
```

Comment:
These will go away if you switch to `framework.TypeDurationSecond` as the framework will handle it for you.